xbmc/cores/VideoPlayer/VideoPlayerAudioID3.cpp

```diff
@@ -0,0 +1,350 @@
+      for (const auto& it : frameListMap)
+      {
+        if (!it.second.isEmpty())
+        {
+          if (it.first == "TIT2")
+          {
+            currentMusic->SetTitle(it.second.front()->toString().to8Bit(true));
+            changed = true;
+          }
+
+          else if (it.first == "TPE1")
+          {
+            currentMusic->SetArtist(GetID3v2StringList(it.second));
+            changed = true;
+          }
+
+          else if (it.first == "TALB")
+          {
+            currentMusic->SetAlbum(it.second.front()->toString().to8Bit(true));
+            changed = true;
+          }
+
+          else if (it.first == "COMM")
+          {
+            // Loop through and look for the main (no description) comment
+            for (const auto& ct : it.second)
+            {
+              auto commentsFrame = dynamic_cast<const ID3v2::CommentsFrame*>(ct);
+              if (commentsFrame && commentsFrame->description().isEmpty())
+              {
+                currentMusic->SetComment(commentsFrame->text().to8Bit(true));
+                changed = true;
+                break;
+              }
+            }
+          }
+
+          else if (it.first == "TCON")
+          {
+            currentMusic->SetGenre(GetID3v2StringList(it.second));
+            changed = true;
+          }
+
+          else if (it.first == "TYER")
+          {
+            currentMusic->SetYear(static_cast<int>(
+                strtol(it.second.front()->toString().toCString(true), nullptr, 10)));
+            changed = true;
+          }
+
+          else if (it.first == "TRCK")
+          {
+            currentMusic->SetTrackNumber(static_cast<int>(
+                strtol(it.second.front()->toString().toCString(true), nullptr, 10)));
+            changed = true;
+          }
+
+          // Support for setting the cover art image via CMusicInfoTag does not currently exist,
+          // the code sample below would check for an ID3v2 "APIC" tag of the proper type and
+          // convert the information into an EmbeddedArt object instance
+          //
+          // else if (it.first == "APIC")
+          // {
+          //  // Loop through and look for the FrontCover picture frame
+          //  for (const auto& pi : it.second)
+          //  {
+          //    auto pictureFrame = dynamic_cast<ID3v2::AttachedPictureFrame*>(pi);
+          //    if (pictureFrame && pictureFrame->type() == ID3v2::AttachedPictureFrame::FrontCover)
+          //    {
+          //      EmbeddedArt coverArt(
+          //          reinterpret_cast<const uint8_t*>(pictureFrame->picture().data()),
+          //          pictureFrame->size(), pictureFrame->mimeType().to8Bit(true));
+
+          //      // Assumes "void CMusicInfoTag::SetCoverArt(const EmbeddedArt& art)" exists
+          //      currentMusic->SetCoverArt(coverArt);
+          //      changed = true;
+          //    }
+          //  }
+          // }
```

Comment:
Yes, the system in Kodi about cover art works currently a bit bad for cases like here. Was also hard to become working here https://github.com/xbmc/xbmc/blob/master/xbmc/addons/AudioDecoder.cpp#L199-L222, as it seems that it contructed for a single playing stream/file and to add them in a gui folder about files and not likes in radio streams where changes on the fly.